src/Microsoft.VisualStudio.Services.Agent/SecretMasker.cs

```diff
@@ -0,0 +1,128 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace Microsoft.VisualStudio.Services.Agent
+{
+    [ServiceLocator(Default = typeof(SecretMasker))]
+    public interface ISecretMasker : IAgentService
+    {
+        IEnumerable<ISecretMask> Masks { get; }
+
+        string MaskSecrets(string input);
+
+        void Add(ISecretMask mask);
+    }
+
+    public class SecretMasker : AgentService, ISecretMasker
+    {
+        private const string MaskedString = "********";
+
+        private readonly List<ISecretMask> _secretMasks = new List<ISecretMask>();
```

Comment:
SecretMasker.MaskSecrets is thread safe. SecretMasker.Add is not thread safe, but it is used in a thread safe way in one place only: Worker.InitializeSecrets.
When we start adding new secrets latter, we need to tackle that problem. The easiest would be do do a lock on "SecretMasker._secretMasks" in "Add" and "MaskSecrets" methods